src/hotspot/cpu/arm/arm.ad

```diff
@@ -10790,7 +10792,7 @@ instruct vsl16B_reg(vecX dst, vecX src, vecX shift) %{
 %}
 
 instruct vsl8B_immI(vecD dst, vecD src, immI shift) %{
-  predicate(n->as_Vector()->length() == 8);
+  predicate(n->as_Vector()->length() == 8 && !n->as_ShiftV()->is_var_shift());
```

Comment:
Is checking for is_var_shift() really necessary for these immI rules?  It seems like it should never happen, so it could be an assert.

Reply:
An assertion should be better, but I didn't find a proper way to add one assertion for `instruct` according to `src/hotspot/share/adlc/Doc/Syntax.doc`.

Hence I'd like to put this check to `predicate`, and "bad AD file` assertion would be issued for unexpected cases.

Reply:
You can put the assert into a helper function.  You can look in aarch64.ad for examples.

bool assert_not_var_shift(const Node *n) {
  assert(!n->as_ShiftV()->is_var_shift(), "illegal var shift");
  return true;
}

[...]

predicate(n->as_Vector()->length() == 8 && assert_not_var_shift(n));

Reply:
Thanks for you suggestion. Will update soon.

Reply:
@dean-long I believe your comments are addressed in the latest commit. Could you please help to take a review when you got a chance? Thanks.

Reply:
Yes, your changes look good.  Approved.